snooty/postprocess.py

```diff
@@ -1669,6 +1671,18 @@ def find_toctree_nodes(
             # Recursively build the tree for each toctree node in this entries list
             for entry in ast.entries:
                 toctree_node: Dict[str, object] = {}
+                if entry.ref_project:
+                    toctree_node = {
+                        "title": [n.Text((0,), entry.title).serialize()]
+                        if entry.title
+                        else None,
+                        "options": {"project": entry.ref_project},
+                    }
+                    if toctree_node in external_nodes:
```

Comment:
If the only methods we call on `external_nodes` are `__contains__()` and `append()`, a set would be more appropriate than a list.

I'm also uncomfortable with the nature of this data structure. It obviously works since everything should be builtin types, but having full dictionaries is harder to reason about and makes typing impossible in our current python version target (3.7; this would be possible in 3.8 and later).

I think I would prefer using a tuple `Tuple[Optional[str], str]` which is what this all collapses to I believe: `(entry.title, entry.ref_project)`

Reply:
very insightful comment! thank you. addressed typing and used a set to initiate (vs a default list)